contracts/src/v0.8/llo-feeds/RewardManager.sol

```diff
@@ -0,0 +1,308 @@
+    _;
+  }
+
+  /// @inheritdoc IRewardManager
+  function onFeePaid(bytes32 poolId, address payee, Common.Asset calldata fee) external override {
+    //this pool only handles LINK fees
+    if (fee.assetAddress != i_linkAddress) {
+      revert InvalidAddress();
+    }
+
+    //update the total fees collected for this pot
+    unchecked {
+      //the total amount for any ERC20 asset cannot exceed 2^256 - 1
+      totalRewardRecipientFees[poolId] += fee.amount;
+    }
+
+    //transfer the fee to this contract
+    IERC20(i_linkAddress).transferFrom(payee, address(this), fee.amount);
+  }
+
+  /// @inheritdoc IRewardManager
+  function claimRewards(bytes32[] memory poolIds) external override {
+    _claimRewards(msg.sender, poolIds);
+  }
+
+  // wrapper impl for claimRewards
+  function _claimRewards(address recipient, bytes32[] memory poolIds) internal returns (uint256) {
+    //get the total amount claimable for this recipient
+    uint256 claimAmount;
+
+    //loop and claim all the rewards in the poolId pot
+    for (uint256 i; i < poolIds.length; ++i) {
+      //get the poolId to be claimed
+      bytes32 poolId = poolIds[i];
+
+      //get the total fees for the pot
+      uint256 totalFeesInPot = totalRewardRecipientFees[poolId];
+
+      unchecked {
+        //get the claimable amount for this recipient, this calculation will never exceed the amount in the pot
+        uint256 claimableAmount = totalFeesInPot - totalRewardRecipientFeesLastClaimedAmounts[poolId][recipient];
+
+        //if there's no fees to claim, continue as there's nothing to update
+        if (claimableAmount == 0) continue;
+
+        //calculate the recipients share of the fees, which is their weighted share of the difference between the last amount they claimed and the current amount in the pot. This can never be more than the total amount in existence
+        uint256 recipientShare = (claimableAmount * rewardRecipientWeights[poolId][recipient]) / PERCENTAGE_SCALAR;
+
+        //keep track of the total amount claimable, this can never be more than the total amount in existence
+        claimAmount += recipientShare;
+
+        //set the current total amount of fees in the pot as it's used to calculate future claims
+        totalRewardRecipientFeesLastClaimedAmounts[poolId][recipient] = totalFeesInPot;
+
+        //emit event if the recipient has rewards to claim
+        emit RewardsClaimed(poolIds[i], recipient, claimAmount);
+      }
+    }
+
+    //check if there's any rewards to claim in the given poolId
+    if (claimAmount > 0) {
+      //transfer the reward to the recipient
+      IERC20(i_linkAddress).transfer(recipient, claimAmount);
+    }
+
+    return claimAmount;
+  }
+
+  /// @inheritdoc IRewardManager
+  function setRewardRecipients(
+    bytes32 poolId,
+    Common.AddressAndWeight[] calldata rewardRecipientAndWeights
+  ) external override onlyOwnerOrFeeManager {
```

Comment:
There was some confusion here since really this should only be set by the `FeeManager`, right? Maybe we should make a more restrictive modifier for it.

Reply:
In most circumstances it would be the FeeManager, but this grants us some flexibility if we may not know the NOP payment address's when setting the configuration, and allows us to add them at a later date. The funds will still be counted for, but can't be claimed until the addresses have been set.